src/com/rcv/Tabulator.java

```diff
@@ -585,14 +592,12 @@ private boolean hasContinuingCandidates(SortedMap<Integer, Set<String>> rankToCa
   // return: map of candidateID to vote tallies for this round
   private Map<String, BigDecimal> getTallyForRound(int currentRound) {
     // map of candidateID to vote tally to store the results
-    Map<String, BigDecimal> roundTally = new HashMap<>();
+    Map<String, BigDecimal> roundTally = getNewTally();
 
-    // initialize round tallies to 0 for all continuing candidates
-    // candidateID indexes through all candidateIDs
-    for (String candidateID : candidateIDs) {
-      if (getCandidateStatus(candidateID) == CandidateStatus.CONTINUING) {
-        roundTally.put(candidateID, BigDecimal.ZERO);
-      }
+    // map of tallies per precinct for this round
+    Map<String, Map<String, BigDecimal>> roundTallyByPrecinct = new HashMap<>();
```

Comment:
Again, should we check if grouping by precincts is specified in the config first before doing this?